crud/common/utils.lua

```diff
@@ -199,18 +199,78 @@ function utils.tarantool_supports_uuids()
     return enabled_tarantool_features.uuids
 end
 
-function utils.convert_operations(user_operations, space_format)
+local function exists_operations_id_map(operations)
+    local map = {}
+    for _, operation in ipairs(operations) do
+        map[operation[2]] = true
+    end
+
+    return map
+end
+
+local function add_nullable_fields_rec(operations, operations_id_map, space_format, tuple, id)
+    if id < 2 or tuple[id - 1] ~= box.NULL then
+        return operations
+    end
+
+    if space_format[id - 1].is_nullable and not operations_id_map[id - 1] then
+        table.insert(operations, {'=', id - 1, box.NULL})
+        return add_nullable_fields_rec(operations, operations_id_map, space_format, tuple, id - 1)
+    end
+
+    return operations
+end
+
+-- Tarantool < 2.3 has no fields `box.error.NO_SUCH_FIELD_NO` and `box.error.NO_SUCH_FIELD_NAME`.
+if _TARANTOOL >= "2.3" then
+    function utils.is_field_not_found(err_code)
+        return err_code == box.error.NO_SUCH_FIELD_NO or err_code == box.error.NO_SUCH_FIELD_NAME
+    end
+else
+    function utils.is_field_not_found(err_code)
+        return err_code == box.error.NO_SUCH_FIELD
+    end
+end
+
+function utils.add_intermediate_nullable_fields(operations, space_format, tuple)
+    if tuple == nil then
+        return operations
+    end
+
     if utils.tarantool_supports_fieldpaths() then
-        return user_operations
+        local formatted_operations, err = utils.convert_operations(operations, space_format)
+        if err ~= nil then
+            return operations
+        end
+
+        operations = formatted_operations
+    end
+
+    -- We need this map to check existence of an operation in constant complexity
+    local operations_id_map = exists_operations_id_map(operations)
+    for i = 1, #operations do
+        operations = add_nullable_fields_rec(
+            operations,
+            operations_id_map,
+            space_format,
+            tuple,
+            operations[i][2]
+        )
     end
 
+    table.sort(operations, function(v1, v2) return v1[2] < v2[2] end)
+    return operations
+end
+
+function utils.convert_operations(user_operations, space_format)
     local converted_operations = {}
 
     for _, operation in ipairs(user_operations) do
         if type(operation[2]) == 'string' then
             local field_id
             for fieldno, field_format in ipairs(space_format) do
-                if field_format.name == operation[2] then
+                local operation_name = operation[2]:split('.', 1)[1]
```

Comment:
It won't work.

Because tuple['["field_name"]'] is appropriate way to get/update tuple field